### PR TITLE
Changes on all plots apply to Colorbar on Tiled Colorfill plots

### LINKED ISSUE #27856: Colorbar incorrectly tied to Tiled Colorfill plots
### Expected behavior
Tiled Colorfill plots should have the colorbar tied to all plots, ie a chnaging the colorbar scale or colormap should change them all!
### Actual behavior
Currently the one colorbar is only tied to the last tile in the plot.
### Steps to reproduce the behavior
Run `CreateSampleWorkspace` twice to get ws1 and ws2
Highlight them both and right-click to produce a colorfill plot.
There is correctly only one colorbar,
but double-click on its scale and edit its limits. The changes will only apply to the second tile as it was the final one plotted. Also various options in the Images tab of the Plot Options menu show that the colorbar is only tied to the last tile
### Platforms affected
Workbench all

### PR DESCRIPTION
**Pull request by alicerussell1**
**Description of work.**
This PR changes changes the settings on colorbars to apply to all the images if there is only one colorbar and multiple plots.

**To test:**
Open workbench and load 2 workspaces (or create 2 using `CreateSampleWorkspace`)
Highlight them both and right-click to produce a colorfill plot.
There should only be one colorbar.
Double-click on its scale and edit its limits. The changes should apply to both images.
Right-click and change the normalization and this should change both images
Open figure options and change any of the settings in the images tab (excluding the image label) and these should apply to all images.

Ensure these settings are still working correctly for single colorfill plots

Fixes #27856 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/applications/workbench/workbench/plotting/propertiesdialog.py [hunk 1]
```diff
@@ -210,16 +208,27 @@ def __init__(self, canvas, axes):
...
-        self.images = self.canvas.figure.gca().images
-        if len(self.images) == 0:
-            self.images = [col for col in self.canvas.figure.gca().collections if isinstance(col, QuadMesh)
-                           or isinstance(col, Poly3DCollection)]
+        self.images=[]
+
+        images = get_images_from_figure(canvas.figure)
+        # If there are an equal number of plots and colorbars so apply changes to plot with the selected colorbar
+        # Otherwise apply changes to all the plots in the figure
+        if len(images) != len(self.canvas.figure.axes)/2:
```
Comment (StephenSmith25) [inline qt/applications/workbench/workbench/plotting/propertiesdialog.py:216]: Bit confused here, so when does it hit the else statement?

**Review by StephenSmith25 — COMMENTED** [review]
Looks good, functions as described, just one suggestion and a comment about something I was confused about.

File: qt/applications/workbench/workbench/plotting/propertiesdialog.py (hunk 1 above)
Comment (alicerussell1) [reply qt/applications/workbench/workbench/plotting/propertiesdialog.py:216]: I think when I was writing it I was thinking if you had more than one colorbar, so say if you had 2 plots each with their own colorbar. It is possible to do this from a script but I don't think anyone does. I can remove it if you think it's unnecessary.

Comment (StephenSmith25) [reply qt/applications/workbench/workbench/plotting/propertiesdialog.py:216]: Ah I see, makes sense. I would say its fine to keep it in there. On the off chance that they create their own colorfill plots it’s probably the behaviour they would desire.

File: Framework/PythonInterface/mantid/plots/datafunctions.py [hunk 2]
```diff
@@ -1039,3 +1040,26 @@ def update_colorbar_scale(figure, image, scale, vmin, vmax):
...
+
+    images = []
+    for ax in axes:
+        images += ax.images + [col for col in ax.collections if isinstance(col, QuadMesh)
+                               or isinstance(col, Poly3DCollection)]
+
+    # remove any colorbar images
+    for img in images:
+        if img.colorbar:
+            images.remove(img.colorbar.solids)
```
Comment (StephenSmith25) [inline Framework/PythonInterface/mantid/plots/datafunctions.py:1056]: Sorry didn't see this the first time, but from what I can see we are modifying the list while iterating over it (I think), which should be undefined behaviour. It appears to work, but might be best to iterate over a copy, or use list comprehension to make a new list.

Comment (alicerussell1) [reply Framework/PythonInterface/mantid/plots/datafunctions.py:1056]: Ah good spot, I had noticed that, I'll change it

**Review by StephenSmith25 — APPROVED** [review]
Looks good to go, good work 👍.

**Pull request merged**